charts/scalardb/values.yaml

```diff
@@ -123,6 +123,22 @@ scalardb:
     # -- The log level of Scalar DB
     dbLogLevel: INFO
 
+  useCustomizedConfiguration:
```

Comment:
Thank you for your review!

> First, using `custom` is probably more appropriate in English.

I see. I will use `custom` as prefix of this parameter. Thank you.

> Second, I think `custom configuration` might be too general, so I think it's better to say it's a custom configuration for `strorageConfiguration`.

Indeed, it is too general. So, I agree with the another name is better.
However, I want to discuss about the word `Storage` is appropriate or not.

After this update, basically we are free to change all settings of Scalar DB Server.
For example, we can change the properties like `scalar.db.server.***` and `scalar.db.consensus_commit.***`.
I feel these are not `storage` configuration.
Also, if some parameters that are not related to `storage` will be implemented in the future, we can change them too.

So, I think `customScalardbConfiguration` is better.

> Lastly, `useXXX` might be misleading since it makes me feel it is a single boolean config.

I agree with you. The `useXXX` looks like boolean config. Thank you.

---

Also, I want to discuss about the name of parameter in the Ledger and Auditor charts.

Before, I thought the same name between several charts is more simple and understandable.
However, I agree with it (`custom configuration`) is too general.
And, according to the same reason (we are free to change all settings), I think it is better to not use the word `storage`.

So, I think the following names are better.

* Scalar DB Server -> `customScalardbConfiguration`
* Ledger -> `customLedgerConfiguration`
* Auditor -> `customAuditorConfiguration`

What do you think?

Reply:
Since there is `storageConfiguration` above in the same file, the name `customStorageConfiguration` sounds like a variant of that object to me.
In this chart, the point of the config is to specify a ConfigMap as a template that will override the properties file path `/scalardb/server/database.properties.tmpl` (which is fixed by the Dockerfile) of the scalardb-server container.
I think a name that could represent this point would be nice, but I haven't come up with one.

Also, we might need to consider that when this mechanism is applied to other charts, the name can be reused or derived naturally.

Reply:
Sorry, I posted my comment without reloading @kota2and3kan san's reply.

Reply:
IMHO, it sounds somewhat inaccurate or non-perfect to me to call this `custom`. Specifying the config values with environment variables in the old way is also "custom", isn't it?

If `useCustomizedConfiguration` is enabled, `storageConfiguration` is completely ignored, right? They are exclusive of each other? If that is true, I wonder if it is a good idea to keep two different ways of configuration. (Not sure if this is a positive suggestion, though.)

Reply:
I see no problem with "Custom" if it provides a default method. If the current default method is to be deprecated in the future, then the name "Custom" becomes a problem.

Whether "storage" should be added depends on whether the configuration file contains only storage. If the configuration file also includes settings other than "storage", then it must be named generically.

Reply:
Another option is to stop using "Custom" by providing default data for the configuration file.

```yaml
scalardb:
  replicaCount: 3
  storageConfiguration:
    contactPoints: cassandra
  configurationData: |
    # Comma separated contact points. For DynamoDB, the region is specified by this parameter.
    scalar.db.contact_points={{ default .Env.SCALAR_DB_CONTACT_POINTS "" }}

    # Port number for all the contact points. Default port number for each database is used if empty.
    scalar.db.contact_port={{ default .Env.SCALAR_DB_CONTACT_PORT "" }}

    # Credential information to access the database. For Cosmos DB, username isn't used. For DynamoDB, AWS_ACCESS_KEY_ID is specified by the username and AWS_ACCESS_SECRET_KEY is specified by the password.
    scalar.db.username={{ default .Env.SCALAR_DB_USERNAME "" }}
    scalar.db.password={{ default .Env.SCALAR_DB_PASSWORD "" }}
```

For example, the above. If users want to customize the configuration file, they can simply edit the default data. This way there is no need for a special field called "Custom".

---

There is another advantage to having the configuration file in values.yaml. If users create their own ConfigMap and write the ConfigMap name in values.yaml, then when users update the configuration file data contained in the ConfigMap, the Pods are not re-created and the configuration file updates are not reflected. The user must explicitly recreate the Pods for the configuration file updates to take effect.

If we choose to write the configuration file data in values.yaml, we can automatically trigger the re-creation of Pods because we will be aware that the data has been updated. There is no need for users to explicitly recreate the Pods.

- https://helm.sh/docs/howto/charts_tips_and_tricks/#automatically-roll-deployments